scipy/stats/_hypotests.py

```diff
@@ -1433,3 +1435,270 @@ def cramervonmises_2samp(x, y, method='auto'):
             p = max(0, 1. - _cdf_cvm_inf(tn))
 
     return CramerVonMisesResult(statistic=t, pvalue=p)
+
+
+class Tukey_HSDResult:
+    """Result of `scipy.stats.tukey_hsd`.
+
+    Attributes
+    ----------
+    statistic : float ndarray
+        The computed statistic of the test for each comparison. The
+        ``(i, j)`` index is the p-value for the ith - jth group comparison.
+    pvalue : float ndarray
+        The associated p-value from the studentized range distribution. The
+        ``(i, j)`` index is the p-value for the ith - jth group comparison.
+
+    Methods
+    -------
+    confidence_interval :
+        Compute the confidence interval for the significance level
+        specified in the test.
+
+
+    References
+    ----------
+    .. [1] NIST/SEMATECH e-Handbook of Statistical Methods, "7.4.7.1. Tukey's
+           Method."
+           https://www.itl.nist.gov/div898/handbook/prc/section4/prc471.htm,
+           28 November 2020.
+    """
+
+    def __init__(self, statistic, pvalue, _nobs, _ntreatments, _stand_err):
+        self.statistic = statistic
+        self.pvalue = pvalue
+        self._ntreatments = _ntreatments
+        self._nobs = _nobs
+        self._stand_err = _stand_err
+
+    def confidence_interval(self, sig_level=.05):
+        """
+        Compute the confidence interval for the specified significance level.
+
+        Parameters
+        ----------
+        confidence_interval : float, optional
+            Confidence level for the computed confidence interval
+            of the estimated proportion. Default is 0.05.
+
+        Returns
+        -------
+        ci : ``ConfidenceInterval`` object
+            The object has attributes ``low`` and ``high`` that hold the
+            lower and upper bounds of the confidence intervals for each
+            comparison. The high and low values are accessible for each
+            comparison at index ``(i, j)`` between groups ``i`` and ``j``..
+
+        References
+        ----------
+        .. [1] NIST/SEMATECH e-Handbook of Statistical Methods, "7.4.7.1.
+               Tukey's Method."
+               https://www.itl.nist.gov/div898/handbook/prc/section4/prc471.htm,
+               28 November 2020.
+        """
+        if not 0 < sig_level < 1:
+            raise ValueError("Significance level must be between 0 and 1.")
+        # determine the critical value of the studentized range using the
+        # appropriate significance level, number of treatments, and degrees
+        # of freedom as determined by the number of data less the number of
+        # treatments. ("Confidence limits for Tukey's method")[1]. Note that
+        # in the cases of unequal sample sizes there will be a criterion for
+        # each group comparison.
+        params = (sig_level, self._nobs, self._ntreatments - self._nobs)
```

Comment:
```suggestion
        params = (1 - confidence_level, self._nobs, self._ntreatments - self._nobs)
```

Reply:
changed to use ppf instead

Reply:
changed to use .ppf instead of doing the `1 - ...`.

Reply:
Ah, good.